Examples/long-to-linked-pe

```diff
@@ -0,0 +1,136 @@
+
+
+def reverse_complement(seq):
+    """Returns reverse complement of the sequence"""
+    return seq[::-1].translate(complements_trans)
+
+
+# From https://github.com/lh3/readfq
+def readfq(fin):  # this is a generator function
+    """Read fasta/fastq file"""
+    last = None  # this is a buffer keeping the last unprocessed line
+    while True:  # mimic closure; is it a bad idea?
+        if not last:  # the first record or a record following a fastq
+            for line in fin:  # search for the start of the next record
+                if line[0] in '>@':  # fasta/q header line
+                    last = line[:-1]  # save this line
+                    break
+        if not last:
+            break
+        name, seqs, last = last[1:].partition(" ")[0], [], None
+        for line in fin:  # read the sequence
+            if line[0] in '@+>':
+                last = line[:-1]
+                break
+            seqs.append(line[:-1])
+        if not last or last[0] != '+':  # this is a fasta record
+            yield name, ''.join(seqs), None  # yield a fasta record
+            if not last:
+                break
+        else:  # this is a fastq record
+            seq, leng, seqs = ''.join(seqs), 0, []
+            for line in fin:  # read the quality
+                seqs.append(line[:-1])
+                leng += len(line) - 1
+                if leng >= len(seq):  # have read enough quality
+                    last = None
+                    yield name, seq, ''.join(seqs)  # yield a fastq record
+                    break
+            if last:  # reach EOF before reading enough quality
+                yield name, seq, None  # yield a fasta record instead
+                break
+
+
+def convert_long_to_linked(in_reads, read_len, fastq):
+    """Convert long reads to pseudo-linked reads"""
+    barcode = 0
+    frag_len = 2 * read_len
+    for name, seq, qual in readfq(in_reads):
+        barcode += 1
+        current_id = name.strip() + "_f{} BX:Z:" + str(barcode)
+        seq = seq.strip()
+
+        if len(seq) < frag_len:
+            continue
+
+        # Split ONT read into fragments of size frag_len
+        read_frags = [seq[i:frag_len + i]
+                      for i in range(0, len(seq), frag_len)]
+
+        if not fastq:
+            fid = 0
+            for frag in read_frags:
+                fid += 1
+                read1 = frag[:read_len]
+                read2 = reverse_complement(frag[-read_len:])
+
+                # Write R1 and R2
+                print(">" + current_id.format(fid) + "\n" + read1)
+                print(">" + current_id.format(fid) + "\n" + read2)
+        else:
+            if not qual:
+                qual = "#" * len(seq)
+
+            # Split quality scores into corresponding fragments of size frag_len
+            qual_frags = [qual[i:frag_len + i] for i in range(0, len(qual), frag_len)]
+
+            fid = 0
+            for read_frag, qual_frag in zip(read_frags, qual_frags):
```

Comment:
Since the quality string is just a dummy string anyway equal to the length of the reads, could we just make it within this for-loop rather than pre-computing and storing all in the list? I know it isn't a big memory issue, but perhaps is using memory unnecessarily?
Something like:
```
qual1 = "#" * len(read1)
```

Reply:
Another option would be to do what @janetxinli in `long-to-linked` in `tigmint` and auto-detect if the input long reads file is fasta or fastq, and use the quality string in the fastq if it's there?

Reply:
Ahh nevermind -- I see now that I missed that you auto-detect the fastq format and use that string if it's there. That makes sense!